File: keepr.client/src/models/Keep.js
import { Account } from "./Account.js";

export class Keep {
  constructor(data) {
    this.id = data.id;
    this.name = data.name;
    this.description = data.description;
    this.img = data.img;
    this.views = data.views;
    this.creator = new Account(data.creator);
    this.kept = data.kept;
    this.createdAt = data.createdAt;
    this.updatedAt = data.updatedAt;
    // this.vaulted=data.vaulted || false
    this.vaultId = data.vaultId || null
    this.vaultKeepId = data.vaultKeepId || null
    this.vaultKeepCreatorId = data.vaultKeepCreatorId || null
  }
}

File: keepr/Models/keep.cs
using System.ComponentModel.DataAnnotations;

namespace keepr.Models;
public class Keep : IHasCreator
{
  public int Id { get; set; }

  [Required]
  public string Name { get; set; }

  public string Description { get; set; }
  [Required]
  public string Img { get; set; }
  public int Views { get; set; } = 0;




  // --------VIRTUALS-------
  public int Kept { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
}

